cdc/log.cc

```diff
@@ -1258,6 +1258,40 @@ class transformer final : public change_processor {
     // Takes and returns generated cdc log mutations and associated statistics about parts touched during transformer's lifetime.
     // The `transformer` object on which this method was called on should not be used anymore.
     std::tuple<std::vector<mutation>, stats::part_type_set> finish() && {
+        static const auto preimg_op_bytes  = _op_col.type->decompose(operation_native_type(operation::pre_image));
+        static const auto postimg_op_bytes = _op_col.type->decompose(operation_native_type(operation::post_image));
+
+        // Adjust or remove deltas according to `delta_mode` in `cdc_options`.
+        if (_schema->cdc_options().get_delta_mode() != cdc::delta_mode::full) {
+            for (auto& m : _result_mutations) {
+                auto& clustered_rows = m.partition().clustered_rows();
+                for (auto it = clustered_rows.begin(); it != clustered_rows.end(); /* no increment */) {
+                    const auto& op_cell = it->row().cells().cell_at(_op_col.id).as_atomic_cell(_op_col);
+                    const auto op_val = op_cell.value().linearize();
+                    if (op_val != preimg_op_bytes && op_val != postimg_op_bytes) {
+                        if (_schema->cdc_options().get_delta_mode() == cdc::delta_mode::off) {
+                            it = m.partition().clustered_rows().erase_and_dispose(it, current_deleter<rows_entry>());
+                            continue;
+                        }
+
+                        if (_schema->cdc_options().get_delta_mode() == cdc::delta_mode::keys) {
```

Comment:
Do we have to check this condition here? Wouldn't an else be enough?

Reply:
We don't have to - I did it for future-proofness (in case we add more delta_mode levels).

Reply:
I don't think this is much more future-proof than just else. If you were throwing when no case is  taken then it would be a different story.